kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/bootstrap/FilterChainFactory.java

```diff
@@ -78,23 +121,35 @@ public ScheduledExecutorService eventLoop() {
                     return pfr.pluginFactory(pluginClass).pluginInstance(instanceName);
                 }
             };
-            this.initialized = filterDefinitions.stream()
-                    .map(fd -> {
-                        FilterFactory<? super Object, ? super Object> filterFactory = pluginFactory.pluginInstance(fd.type());
-                        Class<?> configType = pluginFactory.configType(fd.type());
-                        if (fd.config() == null || configType.isInstance(fd.config())) {
-                            return new UninitializedFilterFactory(fd.type(), filterFactory, fd.config());
-                        }
-                        else {
-                            throw new PluginConfigurationException("accepts config of type " +
-                                    configType.getName() + " but provided with config of type " + fd.config().getClass().getName() + "]");
-                        }
-                    })
-                    .map(uff -> uff.initialize(context))
-                    .toList();
+            this.initialized = new ArrayList<Wrapper>(filterDefinitions.size());
+            try {
+                for (var fd : filterDefinitions) {
+                    FilterFactory<? super Object, ? super Object> filterFactory = pluginFactory.pluginInstance(fd.type());
+                    Class<?> configType = pluginFactory.configType(fd.type());
+                    if (fd.config() == null || configType.isInstance(fd.config())) {
+                        Wrapper uninitializedFilterFactory = new Wrapper(fd.type(), filterFactory, fd.config());
+                        this.initialized.add(uninitializedFilterFactory);
+                        uninitializedFilterFactory.initialize(context);
+                    }
+                    else {
+                        throw new PluginConfigurationException("accepts config of type " +
+                                configType.getName() + " but provided with config of type " + fd.config().getClass().getName() + "]");
+                    }
+                }
+            }
+            catch (Exception e) {
+                // close already initialized factories
+                close();
+                throw e;
+            }
         }
     }
 
+    @Override
+    public void close() {
+        initialized.forEach(Wrapper::close);
```

Comment:
Again I suspect we should be defensive here and continue if `Wrapper::close` throws exceptions.

Reply:
Fixed.